modules/effects/src/effects_metadata.ts

```diff
@@ -40,3 +40,19 @@ export const getSourceMetadata = compose(
   getEffectMetadataEntries,
   getSourceForInstance
 );
+
+export interface EffectsMetadata {
+  [propertyName: string]: {
+    dispatch: boolean;
+  };
+}
+
+export function getEffectsMetadata(instance: any): EffectsMetadata {
```

Comment:
And then the signature of this function would become:

```ts
function getEffectsMetadata<T>(instance: T): EffectsMetadata<T>
```

Reply:
Makes sense, additional type checking is a plus. Not sure it works with the interfaces though, so submitting a change that defines EffectsMetadata as a mapped type.